src/storage.h

```diff
@@ -0,0 +1,260 @@
+
+  ///   @brief generates a request for this facility given its current state. The
+  ///   quantity of the material will be equal to the remaining inventory size.
+  ///   @return a material that this facility will request
+  cyclus::Material::Ptr Request_();
+
+  /// @brief gathers information about bids
+  /// @param commod_requests the materials that have been requested
+  cyclus::BidPortfolio<cyclus::Material>::Ptr GetBids_(
+        cyclus::CommodMap<cyclus::Material>::type& commod_requests, 
+        std::string commod, 
+        cyclus::toolkit::ResourceBuff* buffer);
+
+  /// @brief suggests, based on the buffer, a material response to an offer
+  cyclus::Material::Ptr TradeResponse_(
+      double qty, 
+      cyclus::toolkit::ResourceBuff* buffer);
+
+  /// @brief Move all unprocessed inventory to processing
+  void BeginProcessing_();
+
+  /// @brief Move one ready resource in processing
+  /// @param cap current conversion capacity 
+  void ProcessMat_(double cap);
+
+  /// @brief any ready resources in processing get pushed off to next timestep
+  /// @param time the timestep whose buffer remains unprocessed 
+  void AdvanceUnconverted_(int time);
+
+  /// @brief report the resource quantity in processing at a certain time
+  /// @param time the time of interest
+  double ProcessingAmt_(int time);
+
+  /// @brief this facility's commodity-recipe context
+  inline void crctx(const cyclus::toolkit::CommodityRecipeContext& crctx) {
+    crctx_ = crctx;
+  }
+
+  inline cyclus::toolkit::CommodityRecipeContext crctx() const {
+    return crctx_;
+  }
+
+  /* --- Module Members --- */
+
+  #pragma cyclus var {"tooltip":"input commodity",\
+                      "doc":"commodity accepted by this facility"}
+  std::string in_commod;
+
+  #pragma cyclus var {"tooltip":"output commodity",\
+                      "doc":"commodity produced by this facility"}
+  std::string out_commod;
+
+  #pragma cyclus var {"default":"",\
+                      "tooltip":"input recipe",\
+                      "doc":"recipe accepted by this facility"}
+  std::string in_recipe;
+
+  #pragma cyclus var {"default": 0,\
+                      "tooltip":"process time (timesteps)",\
+                      "doc":"the time it takes to convert a received commodity (timesteps)."}
+  int process_time;
+
+  #pragma cyclus var {"default": 1e299,\
+                     "tooltip":"capacity per timestep (kg)",\
+                     "doc":"the max amount that can be processed per timestep (kg)"}
+  double capacity;
+
+  #pragma cyclus var {"default": 0,\
+                     "tooltip":"cost per kg of production",\
+                     "doc":"cost per kg of produced out_commod"}
+  double cost;
+
+  #pragma cyclus var {"default": 1e299,\
+                      "tooltip":"maximum inventory size (kg)",\
+                      "doc":"the amount of material that can be in storage"}
+  double max_inv_size; 
+
+
+  cyclus::toolkit::ResourceBuff inventory;
```

Comment:
Is the ResBuf variable a user specifiable one? I understood it to be two separate functions, where 'max_inv_size' determines the total amount of material the facility can hold at any time and 'inventory' actually tracks how much and what kind of material is being held in the facility at that time. Or in this case, how much material has been received by the facility and has not been moved into processing yet.